### DISCUSSION TITLE
ST_STM32F429I_DISCOVERY don't work versions upper 1.8.0.759

### DISCUSSION THREAD

**Discussion by maxopa** [nanoframework/Home#1462] [General]
### Target name(s)

ST_STM32F429I_DISCOVERY

### Firmware version

1.8.0.759

### Was working before? On which version?

1.8.0.792

### Description

Hi, i have stmf429i-disc1 board on 32f429zit6u.

I successully flashed last nanoframework firmware ST_STM32F429I_DISCOVERY. I connected the board via microusb cable and get new (COM8) port in dev explorer. The device was visible in VS2022. But on trying to deploy app - VS freezes on erasing 800040, and after rebooting the board new COM8 didn't observed in device explorer(( I tried to downgrade version by version, and on 1.9.0.x after freezing on erasing COM8 device comed in and out every 5 seconds)). It start working on 1.8.0.759. But it don't works with onboard serial ports! I'd get an exception at _port.Open();_. And _SerialPort.GetPortNames();_ shows only COM1 and COM3.
Now i tried to flash it with the last version and i don't see the board on _nanoff --listdevices_, STLink com port is in.

OMG, i can't stand any more. What i'm doing wrong?

**Comment by josesimoes** [comment]
You have a bunch of different issues here... let's move this to a discussion as we're not using issues to answer questions.

**Comment by josesimoes** [comment]
There is nothing wrong with SerialPort.GetPortNames(); showing only COM1 and COM3. Those are the UARTs that are exposed (that's UART0 and UART2)

**Comment by josesimoes** [comment]
Do you have both USB connectors there? Because one is used to power the board and JTAG and the other for USB CDC.

**Reply by maxopa** [reply]
Sure, both are plugged in. But app could be deployed at fw 1.8.0.759 and under. Upper fw version, till 1.9.0.x, deploy stucks (usb cdc com port just going unplugged in windows device manager), 1.9.0.x+ hangs at starting erase memory.

But even at fw 1.8.0.759 i can't work with com ports - exception rises at port.Open() string.

var port = new SerialPort("COM1");
port.Open();

**Comment by josesimoes** [comment]
What is the exception you get?
Are you making any other configurations on the serial por?

**Reply by maxopa** [reply]
I can't lookup it, just exception.

What do you mean other configuration?
First I tried to launch  modbus server using  nanoFramework.Iot.Device.Modbus. but got not implemented exception at server.StartListening(), and after a little investigation I discovered it can't open serial port. I tried to open serial port manually as I shown upper and got exception.

**Reply by maxopa** [reply]
I got exception on opening any serial port, either com1 and com3

**Reply by networkfusion** [reply]
It might be worth trying a new blank program with the following code, and posting the results:

```
var ports = SerialPort.GetPortNames();
Debug.WriteLine("Available serial ports:");
foreach (string port in ports)
{
    try
    {
        Debug.WriteLine($" {port}  TESTING");
        using SerialPort SerialPort = new(port, 9600, Parity.None, 8, StopBits.One);
        SerialPort.Open();
        Thread.Sleep(5000);
        SerialPort.Close();

    }
    catch (Exception ex)
    {
        Debug.WriteLine($"{port}  Exception: {ex}");
    }

}

```

**Reply by networkfusion** [reply]
Also possibily linked to https://github.com/nanoframework/Home/issues/1349

**Reply by maxopa** [reply]
I can't deploy app into controller, VS stucks at erasing deploy memory

System Information
HAL build info: nanoCLR running @ ST_STM32F429I_DISCOVERY
  Target:   ST_STM32F429I_DISCOVERY
  Platform: STM32F4

Firmware build Info:
  Date:        Apr 10 2024
  Type:        MinSizeRel build with ChibiOS v2021.11.3.11
  CLR Version: 1.9.1.47
  Compiler:    GNU ARM GCC v13.2.1

OEM Product codes (vendor, model, SKU): 0, 0, 0

Serial Numbers (module, system):
  00000000000000000000000000000000
  0000000000000000

Target capabilities:
  Has nanoBooter: YES
  nanoBooter: v1.9.1.47
  IFU capable: NO
  Has proprietary bootloader: YES

AppDomains:

Assemblies:

Native Assemblies:
  mscorlib v100.5.0.19, checksum 0x445C7AF9
  nanoFramework.Runtime.Native v100.0.9.0, checksum 0x109F6F22
  nanoFramework.Hardware.Stm32 v100.0.5.1, checksum 0xFE16F347
  nanoFramework.Device.Can v100.0.5.1, checksum 0x5B569036
  nanoFramework.Graphics v100.0.0.10, checksum 0x6A3F90DD
  nanoFramework.Device.OneWire v100.0.4.0, checksum 0xB95C43B4
  nanoFramework.ResourceManager v100.0.0.1, checksum 0xDCD7DF4D
  nanoFramework.System.Collections v100.0.1.0, checksum 0x2DC2B090
  nanoFramework.System.Text v100.0.0.1, checksum 0x8E6EB73D
  nanoFramework.Runtime.Events v100.0.8.0, checksum 0x0EAB00C9
  EventSink v1.0.0.0, checksum 0xF32F4C3E
  System.Math v100.0.5.5, checksum 0x9F9E2A7E
  System.Device.Adc v100.0.0.0, checksum 0xE5B80F0B
  System.Device.Gpio v100.1.0.6, checksum 0x097E7BC5
  System.Device.I2c v100.0.0.2, checksum 0xFA806D33
  System.Device.Pwm v100.1.0.4, checksum 0xABF532C3
  System.IO.Ports v100.1.6.1, checksum 0xB798CE30
  System.Device.Spi v100.1.2.0, checksum 0x3F6E2A7E
  System.Runtime.Serialization v100.0.0.0, checksum 0x0A066871



++++++++++++++++++++++++++++++++
++        Memory Map          ++
++++++++++++++++++++++++++++++++
  Type     Start       Size
++++++++++++++++++++++++++++++++
  RAM   0xd0000000  0x00800000
  FLASH 0x08000000  0x00200000



+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
++                   Flash Sector Map                        ++
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
  Region     Start      Blocks   Bytes/Block    Usage
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
      0    0x08000000       1      0x004000     nanoBooter
      1    0x08004000       3      0x004000     nanoCLR
      2    0x08010000       1      0x010000     nanoCLR
      3    0x08020000       1      0x020000     nanoCLR
      4    0x08040000       6      0x020000     Deployment
      5    0x08100000       4      0x004000     Deployment
      6    0x08110000       1      0x010000     Deployment
      7    0x08120000       7      0x020000     Deployment


+++++++++++++++++++++++++++++++++++++++++++++++++++
++              Storage Usage Map                ++
+++++++++++++++++++++++++++++++++++++++++++++++++++
  Start        Size (kB)           Usage
+++++++++++++++++++++++++++++++++++++++++++++++++++
  0x08000000    0x004000 (16kB)     nanoBooter
  0x08004000    0x03C000 (240kB)    nanoCLR
  0x08040000    0x1C0000 (1792kB)   Deployment



Deployment Map
Empty

14:39:47.875 [Connected to CLR. Pausing execution...]
14:39:48.395 [Erasing sector @ 0x08040000...]
14:41:52.366 [Error erasing sector @ 0x08040000.]
Error erasing ST_STM32F429I_DISCOVERY @ COM8 deployment area.

**Reply by maxopa** [reply]
now it flashed to the last fw version.
